docs/HandlingErrors.md

```diff
@@ -0,0 +1,33 @@
+# Handling errors
+
+## Background
+
+Many different errors can happen at various different points. Having a system to effectively present errors and the details needed to know what to do, and when, is critically important
+
+## Error situations
+
+1. Something broke, and it might be transient, you can try that action again
+1. Some 3rd party dependency failed, and you’ll need to figure out what to do (e.g. GitHub is down, or Xcode couldn't be installed)
+1. You hit a fastlane.ci or fastlane bug
```

Comment:
Maybe we should break up fastlane bugs and fastlane ci bugs, since they have different debugging paths.

Reply:
Yeah, potentially, we could separate it by `fastlane.ci` and `build runner`